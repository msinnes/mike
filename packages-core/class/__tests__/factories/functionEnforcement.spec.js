const functionEnforcementFactory = require('../../src/factories/functionEnforcement');

const enforcement = functionEnforcementFactory('some text');

describe('functionEnforcementFactory', () => {
  it('should be a function', () => {
    expect(functionEnforcementFactory).toBeDefined();
    expect(functionEnforcementFactory).toBeInstanceOf(Function);
  });

  it('should return an enforcement', () => {
    expect(enforcement).toBeDefined();
    expect(enforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error if input is a function', () => {
    expect(() => {
      enforcement(() => {});
    }).not.toThrow();
  });

  it('should throw an error if an input is not a function', () => {
    expect(() => {
      enforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      enforcement({});
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      enforcement();
    }).toThrowErrorMatchingSnapshot();
  });
});
