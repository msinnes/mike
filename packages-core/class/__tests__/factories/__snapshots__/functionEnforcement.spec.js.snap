// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functionEnforcementFactory should throw an error if an input is not a function 1`] = `"InternalError: some text"`;

exports[`functionEnforcementFactory should throw an error if an input is not a function 2`] = `"InternalError: some text"`;

exports[`functionEnforcementFactory should throw an error if an input is not a function 3`] = `"InternalError: some text"`;
